Added linting task

daniel-tabarcea opened the pull request.

daniel-tabarcea committed:
feature(add-liniting-tasks):
- fixed linting issues
- connects unicef/etools-issues#386

daniel-tabarcea committed:
feature(add-linting-fixes):
- added linting task

daniel-tabarcea committed:
Linting fixes

adi130987 commented on package.json:
```diff
@@ -0,0 +1,14 @@
+  "scripts": {
+    "lint": "npm run lint:javascript; exit 0",
+    "lint:javascript": "eslint . --ext js,html --ignore-path .gitignore; exit 0",
+    "//": "polylint\": \"$(npm bin)/polylint -i etools-datepicker-button.html; exit 0"
```
What is the purpose of `"//"`?

daniel-tabarcea replied:
Commented line. Since i can't get polylint to work, i will remove that one.

daniel-tabarcea committed:
feature(add--liniting-tasks): code clean-up

adi130987 committed:
design fix for IE11

fix #heading element height not covering entire left side of the datepicker modal in IE11

daniel-tabarcea committed:
updated linting rules

daniel-tabarcea committed:
updated linting rules

daniel-tabarcea committed:
updated readme with linting instructions
